cli/azd/pkg/infra/azure_resource_manager.go

```diff
@@ -129,10 +130,11 @@ func (rm *AzureResourceManager) GetResourceGroupsForDeployment(
 	// unique set.
 	resourceGroups := map[string]struct{}{}
 
-	for _, dependency := range deployment.Properties.Dependencies {
-		for _, dependent := range dependency.DependsOn {
-			if *dependent.ResourceType == string(AzureResourceTypeResourceGroup) {
-				resourceGroups[*dependent.ResourceName] = struct{}{}
+	for _, resourceId := range deployment.Properties.OutputResources {
```

Comment:
I can also keep the existing logic if we're thinking that it may help in corner-cases. Let me know.